src/services/repo/useRepoBackfilled.tsx

```diff
@@ -0,0 +1,127 @@
+import { useQuery } from '@tanstack/react-query'
+import { useParams } from 'react-router-dom'
+import { z } from 'zod'
+
+import Api from 'shared/api'
+import { NetworkErrorObject } from 'shared/api/helpers'
+import A from 'ui/A'
+
+import {
+  RepoNotFoundErrorSchema,
+  RepoOwnerNotActivatedErrorSchema,
+} from './schemas'
+
+const query = `
+query BackfillFlagMemberships($name: String!, $repo: String!) {
+  config {
+    isTimescaleEnabled
+  }
+  owner(username:$name) {
+    repository(name:$repo) {
+      __typename
+      ... on Repository {
+      flagsMeasurementsActive
+      flagsMeasurementsBackfilled
+      flagsCount
+      }
+      ... on NotFoundError {
+        message
+      }
+      ... on OwnerNotActivatedError {
+        message
+      }
+    }
+  }
+}
+`
+
+const RepositorySchema = z.object({
+  __typename: z.literal('Repository'),
+  flagsMeasurementsActive: z.boolean().nullish(),
+  flagsMeasurementsBackfilled: z.boolean().nullish(),
+  flagsCount: z.number().nullish(),
+})
+
+const RepoBackfilledSchema = z.object({
+  config: z
+    .object({
+      isTimescaleEnabled: z.boolean().nullish(),
+    })
+    .nullish(),
+  owner: z
+    .object({
+      repository: z
+        .discriminatedUnion('__typename', [
+          RepositorySchema,
+          RepoNotFoundErrorSchema,
+          RepoOwnerNotActivatedErrorSchema,
+        ])
+        .nullable(),
+    })
+    .nullable(),
+})
+
+interface URLParams {
+  provider: string
+  owner: string
+  repo: string
+}
+
+export function useRepoBackfilled() {
+  const { provider, owner, repo } = useParams<URLParams>()
```

Comment:
`m`: We've shifted away from grabbing these in the service hooks directly and instead have moved to favouring them being passed in as args, do you mind making that quick update here?

Reply:
Definitely agree with you here, looking at this reference though: https://github.com/codecov/gazebo/blob/9f3692ff90a2a74d2be697f178c724d7b7cdf9d8/src/pages/RepoPage/FlagsTab/BackfillBanners/hooks.ts#L3-L4, we end up going down a rabbit hole increasing the scope of this PR to a degree I'm not super comfortable with. Would prefer to leave as is and tackle in a separate PR

Reply:
<img width="394" alt="Screenshot 2024-05-14 at 9 23 47 AM" src="https://github.com/codecov/gazebo/assets/159853603/0be7021d-deff-42a8-893f-eb2d10bb7be8">
<img width="389" alt="Screenshot 2024-05-14 at 9 24 27 AM" src="https://github.com/codecov/gazebo/assets/159853603/3e2b7ae1-d5bb-4513-9dba-dc0f7ba9e7a1">

Doing a quick search in the codebase, we have 54 hooks to update. I can go ahead and create an epic for that; though I don't really see this getting prioritized